src/ray/core_worker/core_worker.cc

```diff
@@ -281,13 +283,47 @@ void CoreWorker::ReportActiveObjectIDs() {
   heartbeat_timer_.async_wait(boost::bind(&CoreWorker::ReportActiveObjectIDs, this));
 }
 
-void CoreWorker::PromoteObjectToPlasma(const ObjectID &object_id) {
+bool CoreWorker::SerializeObjectId(const ObjectID &object_id, TaskID *owner_id,
+                                   rpc::Address *owner_address) {
   RAY_CHECK(object_id.IsDirectCallType());
   auto value = memory_store_->GetOrPromoteToPlasma(object_id);
   if (value != nullptr) {
     RAY_CHECK_OK(
         plasma_store_provider_->Put(*value, object_id.WithPlasmaTransportType()));
   }
+
+  auto has_owner = reference_counter_->GetOwner(object_id, owner_id, owner_address);
+  if (!has_owner) {
+    // TODO(swang): Once ref counting is fully implemented, this should be an
+    // assertion.
```

Comment:
In what cases could this arise?

Reply:
I believe it should only occur if you are serializing an object ID that was generated randomly. Changing this to an assert.